src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -226,3 +227,25 @@ def update_widget_query(self, query, data, order):
 
     def remove_missing_queries(self, widget_id, keep_ids):
         DashboardWidgetQuery.objects.filter(widget_id=widget_id).exclude(id__in=keep_ids).delete()
+
+
+class DashboardSerializer(DashboardDetailsSerializer):
+    title = serializers.CharField(required=True)
```

Comment:
@markstory turns out `title` is required for new dashboards, and I had to extend the `DashboardDetailsSerializer` to make `title` required without having to be clever.